symtab/src/symtab.rs

```diff
@@ -71,6 +71,10 @@ where
         Ok(())
     }
 
+    pub fn visible_definitions(&self) -> Vec<&S> {
+        self.visible_defs.keys().collect()
```

Comment:
Collecting seems unnecessary to me? You could just return the iterator.
Same goes for the `field_names` getter.

But since we are only calling this in case the compiler aborts because of a detected error, imho this can be merged.

Reply:
I thought of both things too. It was a time tradeoff - I would had to figure out how to return the proper iterator trait :D

Reply:
`... -> impl Iterator<Item=&S>`